File: oyatplum/Week39/과제/같은 숫자는 싫어.js
function solution(arr)
{
    const temp = [];

    for (let i = 0; i < arr.length; i++) {
        if (temp.length == 0) {
            temp.push(arr[i]);
        }
        else {
            const poped = temp.pop();
            if (poped === arr[i]) {
                temp.push(poped);
            }
            else {
                temp.push(poped);
                temp.push(arr[i]);
            }
        }
    }

    return temp;
}

File: un6unw/Week39/algorithm/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        if(arr == null || arr.length == 0){
            return new int[0];
        }

        Stack<Integer> stack = new Stack<>();
        for(int num : arr) {
            if(stack.isEmpty() || stack.peek() != num){
                stack.push(num);
            }
        }
        int size = stack.size();
        int[] answer = new int[size];
        for (int i = size - 1; i >= 0; i--){
            answer[i] = stack.pop();
        }
        return answer;
    }
}